src/client/inputhandler.cpp

```diff
@@ -113,6 +114,19 @@ bool MyEventReceiver::OnEvent(const SEvent &event)
 		return true;
 	}
 
+	if (event.EventType == EET_KEY_INPUT_EVENT) {
+		const KeyPress keyCode(event.KeyInput);
+		if (keyCode == getKeySetting("keymap_fullscreen")) {
+			static bool is_down = false; // used to ignore key repeats
+			if (event.KeyInput.PressedDown && !is_down) {
+				bool fullscreen = RenderingEngine::get_raw_device()->isFullscreen();
+				g_settings->setBool("fullscreen", !fullscreen);
+			}
+			is_down = event.KeyInput.PressedDown;
+			return true;
+		}
+	}
```

Comment:
shouldn't this move below the key handling code and use `keyWasPressed`?

Reply:
That's unfortunately not possible because the key handling code including `keyWasPressed` and stuff only runs when no menu/formspec is open, so the shortcut wouldn't work in the mainmenu. Changing that would probably break other things, so I chose this approach instead.

Reply:
Might be worth adding a comment? As my first thought when reading this was the same as sfan5